Adds scooters and skateboards

PKPenguin321 opened the pull request:
First of all, huge shoutout to https://github.com/tgstation/-tg-station/pull/15427 for both the idea for and a lot of this code. I basically salvaged what @TrustyGun did and touched it up, so he deserves a lot of credit.

**How to make your scooter:**
1. Get 10 rods
2. Craft a scooter frame out of them
3. Apply five metal sheets to make wheels, turning it into a skateboard
4. Apply two rods to make handles, turning the skateboard into a scooter

**How to deconstruct your scooter:**
1. Use a wrench on the scooter to remove the handlebars
2. Use a screwdriver to undo the wheels on the skateboard
3. Use a wrench to undo the scooter frame

_Skateboard specs:_
1. Faster than walking
2. Touching ANYTHING that is dense (people, vending machines, walls, airlocks, anything) will cause you to go careening off of the skateboard, stunning you for a good while and doing 40 stamina damage (which armor cannot block).
3. Very fun, very hilarious

_Scooter specs:_
1. A little bit slower than walking
2. Allows you to move at a decent speed, regardless of any slowdowns applied to you
3. Doesn't crash

:cl: PKPenguin321
rscadd: Scooters and skateboards have been added.
rscadd: Use rods to make a scooter frame. From there, apply metal to make wheels, creating a skateboard. Apply a few more rods to make a full scooter. You can also use tablecrafting to create them, their recipes can be found under the Misc. category.
rscadd: Each step of scooter deconstruction is performed with either a wrench or a screwdriver, depending on the step.
/:cl:

PKPenguin321 committed:
adds scooters and skateboards
adds scooter.dm for pretty much all scooter-and-skateboard related code
added ability to make a scooter frame to rods.dm
added sprites for the scooter, the skateboard, and the scooter frame to vehicles.dmi

PKPenguin321 commented:
what the fugg it compiled locally

Chocobro commented:
S K A T E B O A R D S

PKPenguin321 committed:
updates code to come into line with remie's buckling update

there is still a huge bug that i will point out on github

PKPenguin321 commented on code/modules/vehicles/scooter.dm:
```diff
@@ -0,0 +1,105 @@
+/obj/vehicle/scooter/skateboard/Bump(atom/A)
+	..()
+	if(A.density && buckled_mobs.len)
+		var/mob/living/carbon/human/H = buckled_mobs.len
```
@RemieRichards this line is supposed to be defining "H" as the human riding the vehicle (also on second thought I should change that to just living/carbon not living/carbon/human), but apparently "buckled_mobs.len" is not the human riding the vehicle. I need to know what is, because in its current state this just returns as null and creates loadsa runtimes.

duncathan replied:
buckled_mobs[1] would be a safe bet I think

PKPenguin321 replied:
wtf dunk how did you know that

nevermind @RemieRichards you can thank dunc for taking the work from you

PKPenguin321 commented on code/modules/vehicles/scooter.dm:
```diff
@@ -0,0 +1,105 @@
+					buckled_mob.pixel_y = 4
+				if(WEST)
+					buckled_mob.pixel_x = 0
+					buckled_mob.pixel_y = 4
```
note to self: switch values of south/west directions, they are swapped for some reason

PKPenguin321 committed:
fixes skateboard crashing

PKPenguin321 commented:
>meme PR

no pls i'm serious this time guys

TrustyGun commented:
thanks pk for doing what I could not :+1:

Chocobro commented:
reposting IRC suggestion: elbow/knee pads & helmet for reduced stamina damage when bailing
there's no shame in wearing pads to the park

PKPenguin321 committed:
code cleanup
fixes a case where skateboards can end up in nullspace
adds another do_after
moves a sfx

Cheridan commented:
Manual crafting is mothballed, please do not add any more manual crafting recipes. Use tablecrafting.

Cuboos commented:
If we add blue space crystals... can we get hover boards?

PKPenguin321 commented:
>Manual crafting is mothballed, please do not add any more manual crafting recipes. Use tablecrafting.

having to drag your skateboard off of a table after you're done crafting it is kind of janky and weird and that's why i didn't do it in the first place, but i suppose it shouldn't be hard to whip up a new recipe. i'd like to keep the manual crafting in this instance, however.

make tablecrafting not require a table w h e n ???

PKPenguin321 commented:
kickflips via an action button may come at a later date

PKPenguin321 committed:
adds tablecrafting recipes
also makes skateboards nondense because their sprite was so small and it's awkward bumping into something that small

PKPenguin321 commented:
tablecrafting recipes are in
ready for review

PKPenguin321 committed:
fixes skateboarders being unable to have head on head collisions
since they were non-dense they couldn't crash into each other even when there was another person on the other skateboard. this fixes that

optimumtact commented on code/modules/vehicles/scooter.dm:
```diff
@@ -0,0 +1,114 @@
+	if(istype(I, /obj/item/weapon/screwdriver))
+		user << "<span class='notice'>You begin to deconstruct and remove the wheels on [src]...</span>"
+		playsound(get_turf(user), 'sound/items/Screwdriver.ogg', 50, 1)
+		if(do_after(user, 20, target = src))
```
Is it possible to start the deconstruct step multiple times and get multiple materials

please test.

PKPenguin321 replied:
i do not think that is true because after the first do_after the target qdels leaving no target, meaning the subsequent do_afters would fail
i will test anyways just uuuu but i dont think this is actually an issue

PKPenguin321 replied:
i was right, this is not a real issue

optimumtact replied:
:+1: thanks man

Razharas commented:
>Tablecrafting recipe

Good taste

Razharas commented:
Looks ok btw

HankMoodyExperience commented:
>tablecrafting

nice meme

Cuboos commented:
This is best PR
